runtime/smithy-client/common/src/aws/smithy/kotlin/runtime/client/SdkLogMode.kt

```diff
@@ -70,6 +72,15 @@ public sealed class SdkLogMode(private val mask: Int) {
             LogResponse,
             LogResponseWithBody,
         )
+
+        /**
+         * Parse an [SdkLogMode] from a String
+         * @return the parsed SdkLogMode, or `null` if no log mode could be parsed
+         */
+        public fun fromString(string: String): SdkLogMode? =
+            string.trim().split("|").mapNotNull {
+                allModes().firstOrNull { sdkLogMode -> it.equals(sdkLogMode.toString(), ignoreCase = true) }
+            }.takeIf { it.isNotEmpty() }?.let { it.reduce { acc, sdkLogMode -> acc + sdkLogMode } }
```

Comment:
**Question**: It's strange to me that `fromString` parses a different format than `toString` produces. Should we unify the formats?

Reply:
Yes I think so. I'd rather update the `toString` if nobody opposes that? The current `SdkLogMode(...)` is nice and readable, but it's not intuitive as a configuration value.

Alternatively I could update the `fromString` to support both formats: `SdkLogMode(abc)` and just the inner contents `abc`